ilp-node/src/main/java/org/interledger/node/channels/Channel.java

```diff
@@ -0,0 +1,182 @@
+import org.interledger.node.events.ChannelClosedEvent;
+import org.interledger.node.events.ChannelErrorEvent;
+import org.interledger.node.events.ChannelOpenedEvent;
+import org.interledger.node.events.IncomingRequestEvent;
+import org.interledger.node.events.IncomingTransferEvent;
+import org.interledger.node.exceptions.ChannelNotOpenException;
+import org.interledger.node.handlers.ChannelClosedEventHandler;
+import org.interledger.node.handlers.ChannelErrorEventHandler;
+import org.interledger.node.handlers.ChannelOpenedEventHandler;
+import org.interledger.node.handlers.IncomingRequestHandler;
+import org.interledger.node.handlers.IncomingTransferEventHandler;
+
+import java.io.Closeable;
+import java.util.concurrent.Future;
+
+/**
+ * Defines an abstraction that is meant to be plugged-in to an ILP node system in order
+ * for it to communicate with a another node.
+ *
+ * <p>Based on IL-RFC-24, the general contract for a channels is that, for a single account, it can
+ * send ILP packets and transfers from to the destination node all of which impact the an account
+ * linked to this channel.
+ *
+ * <p>Additionally, channel plugins can also emit events received from the peer, such as incoming
+ * ILP packets or transfers.
+ *
+ * <p>An ILP node will have a unique instance of a channels for each account.
+ *
+ * <p>The following high-level component diagram illustrates how ledger plugins are used by a
+ * connector:
+ *
+ * @see "https://interledger.org/rfcs/0024-ledger-plugin-interface-2/"
+ */
+public interface Channel extends Closeable {
+
+  /**
+   * Called to open the channel.
+   *
+   * <p>This should initiate event subscriptions and establish a connection to the peer.
+   *
+   * <p>Once this method is called, the channels MUST attempt to subscribe to and report events,
+   * including the "open" event immediately after a successful connection is established. If the
+   * connection is lost, the ledger channels SHOULD emit the "closed" event.
+   *
+   */
+  void open();
+
+  /**
+   * Called to disconnect this channel from the peer.
+   *
+   * @throws ChannelNotOpenException if the channels is not connected.
+   *
+   */
+  void close();
+
+  /**
+   * Query whether the channel is currently open.
+   *
+   * @return {@code true} if the channel is open, {@code false} otherwise.
+   */
+  boolean isOpen();
+
+  /**
+   * Initiate a transfer of money to the peer.
+   *
+   * <p>The ILP node will reduce the balance on the account linked to this channels if this call
+   * completes successfully.
+   *
+   * <p>Example: If this channels is used to make transfers over a payment channel then this should
+   * result in a new claim being sent over the channel.
+   *
+   * @param amount The amount to credit the account in favour of the sender.
+   *
+   * @throws ChannelNotOpenException if the channel is not open.
+   *
+   * @return a Future that resolves when the transfer has been sent.
+   */
+  Future<Void> sendTransfer(long amount);
```

Comment:
The naming of this and `sendRequest` are a little confusing/surprising to me, especially after having looked at the [LPI2](https://interledger.org/rfcs/0024-ledger-plugin-interface-2/) for so long.

This operation doesn't _quite_ feel like a transfer (I could be wrong there) -- though I don't really like `sendMoney` either. 

I don't have a good replacement name here, but am wondering what this method is actually doing? I mean, conceptually, "sending money" or even "transferring money from one account to another" makes sense to me (though I would like to be less currency-focused in ILP, so don't like the term `money`). But is this more like clearing or settlement? Or something else? I don't think we should use those terms either (clearing,settle, etc), but am instead trying to figure out -- in layman's terms -- what this operation _does_.

Is it just transferring value from one account to another, across a channel?

Reply:
The node should call this when it has made an adjustment to the balances it is keeping for the account. So imagine the following flow:

  1. Node sends multiple ILP Prepare packets down the Channel to a total value of X.
  2. Balance on the Account for that Channel is now -X.
  3. Node wants to bring the balance back to 0 so it calls `sendTransfer(X)` and adjusts the balance to 0.

As discussed on the call I think we can model this better by separating these functions. I'm going to have an attempt at that today.